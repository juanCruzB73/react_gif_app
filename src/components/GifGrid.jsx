import { useFetchGifs } from "../hooks/useFechGifs"
import { GifGridItem } from "./GifGridItem"
import PropTypes from "prop-types"

export const GifGrid = ({category})=>{

    const{imagenes,isLoading}= useFetchGifs(category)
    
    return(
        <>
        <h3>{category}</h3> 
        {
            isLoading && (<h2>Cargando...</h2>)
        }
        <div className="card-grid">
            {imagenes.map(imagen=>(
                <GifGridItem 
                key={imagen.id}
                {...imagen}//convierte en props lo en lementos de imagen
                />
            ))
            }
        </div>
        </>
    )
}
GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}