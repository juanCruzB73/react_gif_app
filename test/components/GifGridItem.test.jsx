import { render,screen } from "@testing-library/react"
import {GifGridItem} from "../../src/components/GifGridItem"

describe("pruebas en GifGridItem",()=>{
    
    const title = "hola"
    const url = "https://hola.com/"

    test("debe hacer match con snapshot",()=>{ 
        const {container} = render (<GifGridItem title={title} url={url}/>)
        expect(container).toMatchSnapshot();
    })
    test("debe mostrar la imagen con el url y el ALT indicado",()=>{
        render(<GifGridItem title={title} url={url}/>)
        //expect(screen.getByRole("img").src).toBe(url)
        //expect(screen.getByRole("img").alt).toBe(title)
        const {src,alt}=screen.getByRole("img")
        expect(src).toBe(url)
        expect(alt).toBe(title)
    })
    test("debe de mostrar el titulo del componente",()=>{
        render(<GifGridItem title={title} url={url}/>)
        expect(screen.getByText(title)).toBeTruthy()
    })
})