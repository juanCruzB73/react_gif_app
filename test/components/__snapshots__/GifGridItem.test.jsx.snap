// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifGridItem debe hacer match con snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="hola"
      src="https://hola.com/"
    />
    <p>
      hola
    </p>
  </div>
</div>
`;
